dan_layer/wallet/storage_sqlite/src/reader.rs

```diff
@@ -315,6 +315,21 @@ impl WalletStoreReader for ReadTransaction<'_> {
             .collect::<Result<_, _>>()?;
         Ok(outputs)
     }
+
+    fn get_proof_by_transaction_hash(
+        &mut self,
+        transaction_hash: &str,
+    ) -> Result<ConfidentialProofId, WalletStorageError> {
+        use crate::schema::proofs;
+
+        let proof_id = proofs::table
+            .filter(proofs::transaction_hash.eq(transaction_hash))
```

Comment:
When the type is a Hash you can do
```suggestion
            .filter(proofs::transaction_hash.eq(transaction_hash.to_string()))
```